### PR TITLE
Make GitHub pipeline provider to support terraform provisioning

### PR DESCRIPTION
**Pull request by vhvb1989**
fix https://github.com/Azure/azure-dev/issues/622

### CODE REVIEW

File: cli/azd/test/functional/cli_test.go [hunk 1]
```diff
@@ -694,8 +694,6 @@ func newTestContext(t *testing.T) (context.Context, context.CancelFunc) {
 }
 
 func Test_CLI_InfraCreateAndDeleteResourceTerraform(t *testing.T) {
-	// running this test in parallel is ok as it uses a t.TempDir()
```
Comment (ellismg) [inline cli/azd/test/functional/cli_test.go:697]: Did you intend to include this?

**Review by ellismg — APPROVED** [review]
LGTM - I am surprised that we need this, I would have expected that with `AZURE_CREDENTIALS` set, the `az login` action we run as part of the pipleine would mean the az cli could authenticate and then the terraform provider would just use the az CLI's principal.

But if that's not working - this seems like a good fix.

File: cli/azd/test/functional/cli_test.go (hunk 1 above)
Comment (vhvb1989) [reply cli/azd/test/functional/cli_test.go:697]: Yes, terraform tests are inconsistent if running in parallel. @wbreza is looking into it.
But this should help make CI a little more stable

**Review by HadwaAbdelhalem — APPROVED** [review]
Thanks @vhvb1989 ,  LGTM!

File: cli/azd/pkg/commands/pipeline/github_provider.go [hunk 2]
```diff
@@ -305,10 +307,33 @@ func (p *GitHubCiProvider) configureConnection(
...
+		if err := ghCli.SetSecret(ctx, repoSlug, "ARM_TENANT_ID", values.ClientId); err != nil {
+			return fmt.Errorf("setting terraform env var credentials:: %w", err)
+		}
+		if err := ghCli.SetSecret(ctx, repoSlug, "ARM_CLIENT_ID", values.ClientId); err != nil {
+			return fmt.Errorf("setting terraform env var credentials:: %w", err)
+		}
+		if err := ghCli.SetSecret(ctx, repoSlug, "ARM_CLIENT_SECRET", values.ClientId); err != nil {
+			return fmt.Errorf("setting terraform env var credentials:: %w", err)
+		}
+	}
```
Comment (wbreza) [inline cli/azd/pkg/commands/pipeline/github_provider.go:335]: This is a good workaround for now.  We should think about a longer term fix where the we can keep the interaction more loosely coupled.

**Review by wbreza — APPROVED** [review]
Thanks for helping with this :)

**Pull request merged**